e2etest/webhookstests/webhooks.go

```diff
@@ -106,6 +113,50 @@ var _ = ginkgo.Describe("Webhooks", func() {
 			Expect(err).To(HaveOccurred())
 			Expect(err.Error()).To(ContainSubstring("overlaps with already defined CIDR"))
 		})
+
+		ginkgo.DescribeTable("IPAddressPool with overlapping addresses of the nodes",
+			func(ipFamily ipfamily.Family) {
+				nodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				nodeIps, err := k8s.NodeIPsForFamily(nodes.Items, ipFamily, "")
+				Expect(err).NotTo(HaveOccurred())
+				nodeIPString := nodeIps[0]
+				if nodeIPString != "" {
+					nodeIP := net.ParseIP(nodeIPString)
+					cidr := &net.IPNet{
+						IP:   nodeIP,
+						Mask: net.CIDRMask(32, 32),
+					}
+					if nodeIP.To16() != nil {
+						cidr = &net.IPNet{
+							IP:   nodeIP,
+							Mask: net.CIDRMask(128, 128),
+						}
+					}
+					ginkgo.By("Creating IPAddressPool")
+					var nodeCIDRs []string
+					nodeCIDRs = append(nodeCIDRs, cidr.String())
+					resources := config.Resources{
+						Pools: []metallbv1beta1.IPAddressPool{
+							{
+								ObjectMeta: metav1.ObjectMeta{
+									Name: "webhooks-test3",
+								},
+								Spec: metallbv1beta1.IPAddressPoolSpec{
+									Addresses: nodeCIDRs,
+								},
+							},
+						},
+					}
+					err = ConfigUpdater.Update(resources)
+					Expect(err).To(HaveOccurred())
+					Expect(err.Error()).To(ContainSubstring("contains nodeIp"))
+				}
+			},
+			ginkgo.Entry("IPV4", ipfamily.IPv4),
+			ginkgo.Entry("IPV6", ipfamily.IPv6),
+			ginkgo.Entry("DUALSTACK", ipfamily.DualStack),
```

Comment:
I don't think there's value in testing dualstack here (plus nodesip[0] is always ipv4)